# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../vdr-osdteletext/api-vdr-1.5.x.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -aur osdteletext-0.5.1/Makefile osdteletext/Makefile
--- osdteletext-0.5.1/Makefile	2005-08-08 18:29:22.000000000 +0200
+++ osdteletext/Makefile	2006-04-23 17:54:49.000000000 +0200
@@ -20,7 +20,6 @@
 
 ### The directory environment:
 
-DVBDIR = ../../../../DVB
 VDRDIR = ../../..
 LIBDIR = ../../lib
 TMPDIR = /tmp
@@ -32,6 +31,10 @@
 ### The version number of VDR (taken from VDR's "config.h"):
 
 VDRVERSION = $(shell grep 'define VDRVERSION ' $(VDRDIR)/config.h | awk '{ print $$3 }' | sed -e 's/"//g')
+APIVERSION = $(shell grep 'define APIVERSION ' $(VDRDIR)/config.h | awk '{ print $$3 }' | sed -e 's/"//g')
+ifeq ($(strip $(APIVERSION)),)
+  APIVERSION = $(VDRVERSION)
+endif
 
 ### The name of the distribution archive:
 
@@ -40,7 +43,7 @@
 
 ### Includes and Defines (add further entries here):
 
-INCLUDES += -I$(VDRDIR)/include -I$(DVBDIR)/include
+INCLUDES += -I$(VDRDIR)/include
 
 DEFINES += -DPLUGIN_NAME_I18N='"$(PLUGIN)"' -D_GNU_SOURCE
 
@@ -72,7 +75,7 @@
 
 libvdr-$(PLUGIN).so: $(OBJS)
 	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
-	@cp $@ $(LIBDIR)/$@.$(VDRVERSION)
+	@cp $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
 	@-rm -rf $(TMPDIR)/$(ARCHIVE)
Nur in osdteletext: .svn.
diff -aur osdteletext-0.5.1/txtrecv.c osdteletext/txtrecv.c
--- osdteletext-0.5.1/txtrecv.c	2005-08-12 13:20:09.000000000 +0200
+++ osdteletext/txtrecv.c	2007-01-14 00:14:15.667936696 +0100
@@ -692,7 +692,9 @@
 */
 
 cTxtReceiver::cTxtReceiver(int TPid, tChannelID chan)
-#if VDRVERSNUM >= 10319
+#if VDRVERSNUM >= 10500
+ : cReceiver(chan, -1, TPid), 
+#elif VDRVERSNUM >= 10319
  : cReceiver(0, -1, TPid), 
 #else
  : cReceiver(0, -1, 1, TPid), 
